python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -2476,7 +2476,13 @@ def repeat(col: "ColumnOrName", n: Union["ColumnOrName", int]) -> Column:
 repeat.__doc__ = pysparkfuncs.repeat.__doc__
 
 
-def split(str: "ColumnOrName", pattern: str, limit: int = -1) -> Column:
+def split(
+    str: "ColumnOrName",
```

Comment:
We might want to consider renaming variable that shadows python built-in such as `str`. It's annoying here that it breaks the usage of `isinstance(pattern, str)`, and I need to work around this by aliasing the python builtin.
Given renaming variable would be a backwards incompatible change (for user code that uses kwargs) I won't add that here.